lib/nexthop.h

```diff
@@ -57,6 +57,7 @@ struct nexthop {
 
 	/* Interface index. */
 	ifindex_t ifindex;
+	char ifname[INTERFACE_NAMSIZ + 1];
```

Comment:
I'd strongly appreciate if we could not do this and contain the interface name logic in the static route handling code.